RELEASE_NOTES.md

```diff
@@ -20,6 +20,26 @@
 
 * Now exceptions are not raised in Receiver.ready() but in Receiver.consume() (receive() or the async iterator `anext`).
 
+* `Select` constructor now takes a variable number of receivers:
+
+  ```py
+  select = Select(recv1, recv2)
+  ```
+
+* `Select.ready()` is now an async iterator and yields a set of receivers that are ready to be consumed. Receivers must be explicitly consumed and if a ready receiver is not consumed, the ready message won't be discarded by select any more, it will wait indefinitely until it is consumed.
+
+  Example:
+
+  ```py
+  select = Select(recv1, recv2)
+  async for ready_set in select.ready():
+      if recv1 in ready_set:
+          msg = recv1.consume()
```

Comment:
Users having to call `consume` directly make me a bit uneasy.  I would much rather advertise using `await recv.receive()` here,  because all our receiver implementations guarantee that until a previous value is read, subsequent calls to `ready()` don't overwrite unread values in the receiver.

That way, normal users only have to keep one thing `receive()` in mind,  or they can start doing `val = await anext(recv)` from python 3.10.

Reply:
Why does it make you uneasy? I don't see the difference between using `await recv.receive()` or `recv.consume()` from the safety POV, if you got your receiver from the `ready_set`, it is guaranteed that `ready()` was called, so there is something to consume. Even more, it is guaranteed that you won't get blocked when reading the next value, which IMHO is a nice property to have, you know the entire loop won't block once receivers are ready. I don't see why one would like to await for something that it is guaranteed to be there.

Am I missing or misunderstanding anything?

Reply:
This is because it is very important to provide a single simple way of doing things,  that's the Python way.

You have many months worth of familiarity with this code, but for someone who is new to this,  it will look like a mysterious thing,  and they'll have to read the documentation to read what `consume` is,  and they'll have to make sense of the design etc, whereas all they wanted was a way to quickly get a `select`.

> it is guaranteed that you won't get blocked when reading the next value

the guarantee remains if we call `receive` on a ready channel as well.

Reply:
> This is because it is very important to provide a single simple way of doing things, that's the Python way.

I agree with this, but I think looping an async iterator is a different thing that `await`ing a single message. The whole point of an async iterator is that it does the `await` for you.

> You have many months worth of familiarity with this code, but for someone who is new to this, it will look like a mysterious thing, and they'll have to read the documentation to read what consume is, and they'll have to make sense of the design etc, whereas all they wanted was a way to quickly get a select.

For me it is not that mysterious because one is async (`receive()` and the other one is sync (`consume()`), that's what makes it much clear IMHO, if both were sync or async I agree it would be 2 ways to do the same thing.

> the guarantee remains if we call `receive` on a ready channel as well.

Yes, it is just less efficient because you are adding unnecessary context switches.

Also, nobody is preventing users from call `await recv.receive()` instead, and it will work and do what the user expects, so I don't see a big issue here.

In any case, if this is about documentation or how we present examples, we could explain more explicitly why `consume()` is recommended (which I think we should recommend anyway) and mention that of course `await recv.receive()` will work too.

Reply:
If that is acceptable for you I can improve the examples and docs.